core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -691,39 +703,62 @@ private void freeSpaceInternal(long sessionId, long availableBytes, BlockStoreLo
       }
       blocksGroupedByDestTier.get(alias).add(entry);
     }
+
     // 2.2. move blocks in the order of their dst tiers, from bottom to top
     for (int tierOrdinal = mStorageTierAssoc.size() - 1; tierOrdinal >= 0; --tierOrdinal) {
       Set<BlockTransferInfo> toMove =
           blocksGroupedByDestTier.get(mStorageTierAssoc.getAlias(tierOrdinal));
       if (toMove == null) {
         toMove = new HashSet<>();
       }
+      // Create move tasks for current tier.
+      List<Callable<Boolean>> moveTasks = new ArrayList<>(toMove.size());
       for (BlockTransferInfo entry : toMove) {
-        long blockId = entry.getBlockId();
-        BlockStoreLocation oldLocation = entry.getSrcLocation();
-        BlockStoreLocation newLocation = entry.getDstLocation();
-        MoveBlockResult moveResult;
-        try {
-          moveResult = moveBlockInternal(Sessions.createInternalSessionId(),
-              blockId, oldLocation, newLocation);
-        } catch (InvalidWorkerStateException e) {
-          // Evictor is not working properly
-          LOG.error("Failed to demote blockId {}, this is temp block", blockId);
-          continue;
-        } catch (BlockAlreadyExistsException e) {
-          continue;
-        } catch (BlockDoesNotExistException e) {
-          LOG.info("Failed to demote blockId {}, it could be already deleted", blockId);
-          continue;
-        }
-        if (moveResult.getSuccess()) {
-          synchronized (mBlockStoreEventListeners) {
-            for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
-              listener.onMoveBlockByWorker(sessionId, blockId, moveResult.getSrcLocation(),
-                  newLocation);
+        moveTasks.add(() -> {
+          long blockId = entry.getBlockId();
+          BlockStoreLocation oldLocation = entry.getSrcLocation();
+          BlockStoreLocation newLocation = entry.getDstLocation();
+
+          try {
+            MoveBlockResult moveResult = moveBlockInternal(Sessions.createInternalSessionId(),
+                blockId, oldLocation, newLocation);
+
+            if (moveResult.getSuccess()) {
+              synchronized (mBlockStoreEventListeners) {
+                for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
+                  listener.onMoveBlockByWorker(sessionId, blockId, moveResult.getSrcLocation(),
+                      newLocation);
+                }
+              }
+              return true;
             }
+          } catch (InvalidWorkerStateException e) {
+            // Evictor is not working properly
+            LOG.error("Failed to demote blockId {}, this is temp block", blockId);
```

Comment:
Should this be warn or should we do something to handle this error?

Reply:
This is not expected so I'd leave it ERROR based on above comment.